react/ProductSummaryBuyButton.tsx

```diff
@@ -1,44 +1,60 @@
 import React from 'react'
-import PropTypes from 'prop-types'
-import BuyButton from 'vtex.store-components/BuyButton'
-import { withRuntimeContext } from 'vtex.render-runtime'
+import { BuyButton } from 'vtex.store-components'
+import { useRuntime } from 'vtex.render-runtime'
 // eslint-disable-next-line no-restricted-imports
-import { path, equals } from 'ramda'
+import { equals } from 'ramda'
 import classnames from 'classnames'
 import { IOMessage } from 'vtex.native-types'
-import { useProductSummary } from 'vtex.product-summary-context/ProductSummaryContext'
+import { ProductSummaryContext } from 'vtex.product-summary-context'
 import { useCssHandles } from 'vtex.css-handles'
 
 import displayButtonTypes, {
   getDisplayButtonNames,
   getDisplayButtonValues,
-} from '../../utils/displayButtonTypes'
-import productSummary from '../../productSummary.css'
+} from './utils/displayButtonTypes'
+import productSummary from './productSummary.css'
 
-const ALWAYS_GO_TO_PRODUCT = 'alwaysGoToProduct'
 const ALWAYS_ADD_TO_CART = 'alwaysAddToCart'
 const DEFAULT_BUTTON_BEHAVIOR = 'default'
-const BUY_BUTTON_BEHAVIOR_OPTIONS = [
-  ALWAYS_GO_TO_PRODUCT,
-  ALWAYS_ADD_TO_CART,
-  DEFAULT_BUTTON_BEHAVIOR,
-]
 
-const CSS_HANDLES = ['buyButton', 'buyButtonContainer']
+const CSS_HANDLES = ['buyButton', 'buyButtonContainer'] as const
 
-const ProductSummaryBuyButton = ({
-  displayBuyButton,
-  isOneClickBuy,
+interface Props {
+  /** What the buy button should do when you click it, if you pass default it will add to cart only if there is only one SKU of that product */
+  buyButtonBehavior?: 'default' | 'alwaysGoToProduct' | 'alwaysAddToCart'
+  /** Should redirect to checkout after clicking on buy */
+  isOneClickBuy?: boolean
+  /** Custom buy button text */
+  buyButtonText?: string
+  /**
+   * Defines the display mode of buy button
+   * @default "displayButtonAlways"
+   */
+  displayBuyButton?:
+    | 'displayButtonAlways'
+    | 'displayButtonHover'
+    | 'displayButtonNone'
+  /** A custom URL for the `VIEW CART` button inside the toast created by BuyButton */
+  customToastURL?: string
+  isHovering?: boolean
+}
+
+function ProductSummaryBuyButton({
+  displayBuyButton = 'displayButtonAlways',
+  isOneClickBuy = false,
   buyButtonText,
   customToastURL,
-  runtime: {
-    hints: { mobile },
-  },
-  buyButtonBehavior,
+  buyButtonBehavior = DEFAULT_BUTTON_BEHAVIOR,
   isHovering,
-}) => {
-  const { product, selectedItem, selectedQuantity } = useProductSummary()
-  const handles = useCssHandles(CSS_HANDLES)
+}: Props) {
+  const {
+    hints: { mobile },
+  } = useRuntime()
+
+  const { product, selectedItem, selectedQuantity } =
+    ProductSummaryContext.useProductSummary() ?? {}
+
+  const { handles } = useCssHandles(CSS_HANDLES)
```

Comment:
Shouldn't this component have `classes` too?

Reply:
It's deprecated